sunpy/map/header_helper.py

```diff
@@ -361,3 +361,52 @@ def _set_instrument_meta(meta_wcs, instrument, telescope, observatory, wavelengt
     'CD2_2':
     'Matrix element CDi_j describing the rotation required to align solar North with the top of the image.'
 }
+
+
+def carrington_header(date, observer_coordinate, *, shape_out, projection_code="CAR"):
```

Comment:
```suggestion
def carrington_header(date, observer_coordinate, *, dimensions, projection_code="CAR"):
```
I suggest changing the name for this keyword (here and thereafter) to make it clear that it follows our "dimensions" ordering rather than the NumPy "shape" ordering

Reply:
Since I swapped the dimensions round to be consistent with the general header helper, I think this is fine now.